### PR TITLE
Strongly type ids

### PR DESCRIPTION
**Pull request by fitzgen**
* [X] `TypeId`
* [X] `ModuleId`
* [x] `VarId`
* [x] `FunctionId`

### CODE REVIEW

File: src/ir/context.rs [hunk 1]
```diff
@@ -1066,15 +1253,15 @@ impl BindgenContext {
     }
 
     /// Look up whether the item with `id` has vtable or not.
-    pub fn lookup_item_id_has_vtable(&self, id: &ItemId) -> bool {
+    pub fn lookup_item_id_has_vtable<Id: Into<ItemId>>(&self, id: Id) -> bool {
```
Comment (pepyakin) [inline src/ir/context.rs]: I suppose `has_vtable` is defined only for types and calling it on a not type is bit smelly. Would it be hard to change this fn to only accept `TypeId`?

Comment (fitzgen) [reply src/ir/context.rs]: Good call -- I'll give it a try.

**Comment by fitzgen** [comment]
@pepyakin OK, I made `is_unsized`, `has_vtable`, and `has_destructor` methods all take `TypeId`.

**Comment by pepyakin** [comment]
Looks great!

@bors-servo r+

**Comment by fitzgen** [comment]
@bors-servo r=pepyakin

Hmm doesn't seem to be deployed yet...

**Pull request merged**